### PR TITLE
Adds short tutorial to readme and simple example, improves package docs

### LINKED ISSUE #78: Short docs on how to use OpenPubkey library
This documentation should consist of:
1. A short paragraph length description in the readme for how use OpenPubkey as a library
2. and a 4 page markdown document that walks a developer through getting started with OpenPubkey

### PR DESCRIPTION
**Pull request by EthanHeilman**
This PR is written to address and close #78  by adding a short "Getting Started" tutorial to the readme. 

* To support this tutorial this PR also adds a very simple signing and verifying example and improves package docs on functions used in the tutorial.

* This PR also cleans up most of the markdown warnings in the README and adds `typ` claim to the commitment

This simple example will likely slowly replace the google example. Future work is to merge the Google example with the X509 example to create a general purpose signing utility. The creation this signing utility will be done in a future PR and will not be done in this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,47 @@ Companies building on OpenPubkey include:
...
+  RedirURIPort: "3000",
+  CallbackPath: "/login-callback",
+  RedirectURI:  "http://localhost:3000/login-callback",
+}
+```
+
+Next we create the OpenPubkey client and call `opkClient.Auth`:
+```golang
+opkClient, err := client.New(op)
+pkt, err := opkClient.Auth(context.TODO())
```
Comment (whalelines) [inline README.md:39]: Should some comment be made about `context.TODO()`?

**Review by whalelines — COMMENTED** [review]
Thanks for adding the godoc comments.

File: README.md (hunk 1 above)
Comment (EthanHeilman) [reply README.md:39]: What do you suggest?

Comment (whalelines) [reply README.md:39]: What is it and why is it called `TODO`? What is to be done?

Comment (EthanHeilman) [reply README.md]: Changed this to use `context.Background()` instead. I want to get developers up and running with requiring that make as few choices as possible. `context.TODO()` implies that should replace the context with something else. Instead I'd rather give them a sane default.

Comment (EthanHeilman) [reply README.md]: Should be fixed

**Review by whalelines — APPROVED** [review]
LGTM

**Pull request merged**